client/app/bundles/course/level/pages/Level/index.jsx

```diff
@@ -0,0 +1,191 @@
+    id: 'course.level.Level.addNewLevel',
+    defaultMessage: 'Add New Level',
+  },
+  saveLevels: {
+    id: 'course.level.Level.saveLevels',
+    defaultMessage: 'Save Levels',
+  },
+  saveSuccess: {
+    id: 'course.level.Level.saveSuccess',
+    defaultMessage: 'Levels Saved',
+  },
+});
+
+const styles = {
+  body: {
+    display: 'flex',
+  },
+  sidebar: {
+    width: 250,
+  },
+  mainPanel: {
+    paddingLeft: 40,
+    paddingRight: 40,
+    width: '100%',
+  },
+  duplicateButton: {
+    display: 'flex',
+    justifyContent: 'center',
+  },
+  formButton: {
+    marginRight: 10,
+  },
+};
+
+class Level extends React.Component {
+  static propTypes = {
+    isLoading: PropTypes.bool.isRequired,
+    levels: PropTypes.arrayOf(PropTypes.number).isRequired,
+
+    dispatch: PropTypes.func.isRequired,
+  }
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+    };
+
+    this.handleUpdateExpThreshold = this.handleUpdateExpThreshold.bind(this);
+    this.handleLevelTextBlur = this.handleLevelTextBlur.bind(this);
+    this.handleCreateLevel = this.handleCreateLevel.bind(this);
+    this.handleDeleteLevel = this.handleDeleteLevel.bind(this);
+    this.handleSaveLevels = this.handleSaveLevels.bind(this);
+  }
+
+  componentDidMount() {
+    this.props.dispatch(fetchLevels());
+  }
+
+  handleUpdateExpThreshold(levelNumber, newValue) {
+    this.props.dispatch(updateExpThreshold(levelNumber, newValue));
+  }
+
+  handleLevelTextBlur() {
+    this.props.dispatch(sortLevels());
+  }
+
+  handleCreateLevel() {
+    return (e) => {
+      e.preventDefault();
+      if (!this.props.isLoading) {
+        this.props.dispatch(addLevel());
+      }
+    };
+  }
+
+  handleDeleteLevel(levelNumber) {
+    return (e) => {
+      e.preventDefault();
+      if (!this.props.isLoading) {
+        this.props.dispatch(deleteLevel(levelNumber));
+      }
+    };
+  }
+
+  handleSaveLevels() {
+    return (e) => {
+      e.preventDefault();
+      const successMessage = <FormattedMessage {...translations.saveSuccess} />;
+      if (!this.props.isLoading) {
+        this.props.dispatch(saveLevels(this.props.levels, successMessage));
+      }
+    };
+  }
+
+  renderBody() {
+    const rows = this.props.levels.slice(1).map((experiencePointsThreshold, index) => (
+      <LevelRow
+        key={index + 1}
+        levelNumber={index + 1}
+        experiencePointsThreshold={experiencePointsThreshold}
+        updateExpThreshold={this.handleUpdateExpThreshold}
+        sortLevels={this.handleLevelTextBlur}
+        deleteLevel={this.handleDeleteLevel}
+        isLoading={this.props.isLoading}
+      />
+    ));
+
+    return (
+      <div style={styles.body}>
+        <Table className="table levels-list">
+          <TableHeader adjustForCheckbox={false} displaySelectAll={false}>
+            <TableRow>
+              <TableHeaderColumn style={{ paddingLeft: '74px' }}><FormattedMessage {...translations.levelHeader} /></TableHeaderColumn>
```

Comment:
Would prefer if we shifted most of the style attributes up into the `styles` const.

Reply:
Ok done!